lib/chef/client.rb

```diff
@@ -555,6 +570,32 @@ def save_updated_node
       end
     end
 
+    #
+    # Populate the minimal ohai attributes defined in #run_ohai with data train collects.
+    #
+    # Eventually ohai may support colleciton of data.
+    #
+    def get_ohai_data_remotely
+      ohai.data[:fqdn] = if transport_connection.respond_to?(:hostname)
+                           transport_connection.hostname
+                         else
+                           Chef::Config[:target_mode][:host]
+                         end
+      if transport_connection.respond_to?(:os)
+        ohai.data[:platform] = transport_connection.os.name
+        ohai.data[:platform_version] = transport_connection.os.release
+        ohai.data[:os] = transport_connection.os.family_hierarchy[1]
+        ohai.data[:platform_family] = transport_connection.os.family
```

Comment:
I'm not sure if we want to do this actually. These platforms aren't the same and populating attributes normally handled by ohai with data from train is going to cause a bad user experience. Many of the platforms are not the same so you'd write a cookbook expecting one behavior and you'd get another.

Reply:
Take a peak at the work that @burtlo did to translate train to ohai values.

Reply:
I think we have to do this for the platform mapping to not blow up on us, probably other parts of code expect `node[:platform]` to be set. Otherwise `node[:train][:os][:name]` would be logical.

Of course I agree we still need to reconcile ohai vs train data here. or https://github.com/burtlo/ohbye, or similar.

We've got a ways to go before you could use the same cookbooks for a target mode and normal node though.